traits/ctrait.py

```diff
@@ -109,6 +111,38 @@ def comparison_mode(self):
     def comparison_mode(self, value):
         ctraits.cTrait.comparison_mode.__set__(self, value)
 
+    def property(self, *args):
```

Comment:
This isn't the API that we want: in general, it's bad design to have a function or method do different things depending on the type of arguments (or the number of arguments) that it's called with - that's usually a sign that you actually want two separate functions or methods. (This was the main point of issue #825.)

Instead, as discussed in the [PR #837 comments](https://github.com/enthought/traits/pull/837#issuecomment-603834739), I think we want a Python `property` here. Though I can see that creating a property called `property` may require some finessing.

Reply:
Oh, I misunderstood that comment, yes, this didn't feel right to me either, will try to create a Python `property` for property.

Reply:
Turns out, it doesn't cause any namespace conflict, I had made a coding error previously, didn't have to do anything special to make it a python property.